components/notebook-controller/Makefile

```diff
@@ -1,18 +1,10 @@
 
 # Image URL to use all building/pushing image targets
-IMG ?= gcr.io/kubeflow-images-public/notebook-controller
+IMG ?= public.ecr.aws/j1r0q0g6/notebooks/notebook-controller
 TAG ?= $(shell git describe --tags --always)
-SHELL := /bin/bash
-GOLANG_VERSION ?= 1.15
 
-# Whether to use cached images with GCB
-USE_IMAGE_CACHE ?= true
-
-# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
-CRD_OPTIONS ?= "crd"
```

Comment:
Thanks for the comments! I'm starting to review them.

Is `kubectl replace` instead of `kubectl apply` an option? It should fix both errors:

```
kubectl replace -f config/crd/bases/kubeflow.org_notebooks.yaml
```

Reply:
After looking at this again I propose that we manually 
1. add the `spec.preserveUnknownFields: false` just for this release
2. remove this field in KF 1.7, since it's only needed for the `apiextensions.k8s.io/v1beta1` to `apiextensions.k8s.io/v1` transition.

The `controller-gen` does not allow us to explicitly set this to false https://book.kubebuilder.io/reference/markers/crd-processing.html

Reply:
I have added it to the `config/crd/patches/trivial_conversion_patch.yaml` file instead of using `yq` or `sed` hacks:

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: notebooks.kubeflow.org
spec:
  preserveUnknownFields: false # TODO: Remove in Kubeflow 1.7 release
  conversion:
    strategy: None
```

Reply:
This is awesome @samuelvl!

One final nit, I see that you've commented out the `crd` part in the manifests, which results in the CRD to not be included when generating `base` or `overlays/kubeflow` 

https://github.com/kubeflow/kubeflow/pull/6374/files#diff-28481732533c7c75d2d2ba504e9670d90e72ed98f999115fd92a0f8e8a5aace2R20

Could you revert it back, so that the CRD is included as well? We should be ready to merge afterwards

Reply:
I commented the `crd` part to avoid this issue (https://github.com/prometheus-community/helm-charts/issues/1500 is similar): 

``` The CustomResourceDefinition "notebooks.kubeflow.org" is invalid: metadata.annotations: Too long: must have at most 262144 bytes ``` 

I have reverted the original logic and added the `maxDescLen=0` option when generating the crd to avoid this issue, however we'll miss some information when using the `kubectl explain notebook` command:

``` .PHONY: manifests
manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
	$(CONTROLLER_GEN) rbac:roleName=role crd:maxDescLen=0 ... ```

Reply:
Hmm, indeed in this PR we made the CRD quite bigger since we now actually include the full PodSpec for validation.

But the `maxDescLen=0` seems like a good way to cut down significantly the size of it. The spec is currently only the PodSpec, for which descriptions are widely available so we are OK.

Good job @samuelvl!